.github/workflows/reusable-api-e2e.yml

```diff
@@ -14,83 +14,47 @@ jobs:
-      id: pnpm-install
-      with:
-        version: 7.5.0
-        run_install: false
-
-    - name: Get pnpm store directory
-      id: pnpm-cache
-      shell: bash
-      run: |
-        echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
-
-    - uses: actions/cache@v3
-      name: Setup pnpm cache
-      with:
-        path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
-        key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
-        restore-keys: |
-          ${{ runner.os }}-pnpm-store-
-
-    - name: Install dependencies
-      run: pnpm install
-
-    - uses: mansagroup/nrwl-nx-action@v3
-      with:
-        targets: lint
-        projects: "@novu/api"
-
-    - name: Start Local Stack
-      env:
-        AWS_DEFAULT_REGION: us-east-1
-        DEFAULT_REGION: us-east-1
-        AWS_ACCOUNT_ID: '000000000000'
-        AWS_ACCESS_KEY_ID: test
-        AWS_SECRET_ACCESS_KEY: test
-        AWS_EC2_METADATA_DISABLED: true
-      working-directory: docker/local
-      run: |
-        docker-compose -f docker-compose.localstack.yml up -d
-        sleep 10
-        max_retry=30
-        counter=0
-        until $command
-        do
-          sleep 1
-          [[ counter -eq $max_retry ]] && echo "Failed!" && exit 1
-          aws --endpoint-url=http://localhost:4566 s3 ls
-          echo "Trying again. Try #$counter"
-          ((counter++))
-        done
-        docker-compose -f docker-compose.localstack.yml logs --tail="all"
-        aws --endpoint-url=http://localhost:4566 --cli-connect-timeout 600 s3 mb s3://novu-test
-
-      # Runs a single command using the runners shell
-    - name: Build API
-      run: CI='' pnpm build:api
-
-    # Runs a set of commands using the runners shell
-    - name: Run a test
-      run: |
-        cd apps/api && pnpm test:e2e
-        pnpm test
-
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/setup-project
+
+      - uses: mansagroup/nrwl-nx-action@v3
+        with:
+          targets: lint
+          projects: "@novu/api"
+
+      - name: Start Local Stack
+        env:
+          AWS_DEFAULT_REGION: us-east-1
+          DEFAULT_REGION: us-east-1
+          AWS_ACCOUNT_ID: '000000000000'
+          AWS_ACCESS_KEY_ID: test
+          AWS_SECRET_ACCESS_KEY: test
+          AWS_EC2_METADATA_DISABLED: true
+        working-directory: docker/local
+        run: |
+          docker-compose -f docker-compose.localstack.yml up -d
+          sleep 10
+          max_retry=30
+          counter=0
+          until $command
+          do
+            sleep 1
+            [[ counter -eq $max_retry ]] && echo "Failed!" && exit 1
+            aws --endpoint-url=http://localhost:4566 s3 ls
+            echo "Trying again. Try #$counter"
+            ((counter++))
+          done
+          docker-compose -f docker-compose.localstack.yml logs --tail="all"
+          aws --endpoint-url=http://localhost:4566 --cli-connect-timeout 600 s3 mb s3://novu-test
```

Comment:
Should we set the localhost url and port as env variable and provide it to the workflow?

Reply:
Yes most defiently, There is another round of refactoring here to be made. Left this as is for now as out of the scope for this PR